src/LightInject.Interception/LightInject.Interception.cs

```diff
@@ -1807,8 +1807,17 @@ private MethodBuilder ImplementInterceptedMethod(MethodInfo targetMethod, int[]
                 
             if (proxyDefinition.TargetType.GetTypeInfo().IsInterface && proxyDefinition.ImplementingType != null)
             {
-                targetInvocationMethod = proxyDefinition.ImplementingType.GetTypeInfo().DeclaredMethods.FirstOrDefault(m => m.Name == targetMethod.Name);
-            }            
+                var type = proxyDefinition.ImplementingType;
+
+                while (type != null)
+                {
+                    targetInvocationMethod = type.GetTypeInfo().DeclaredMethods.FirstOrDefault(m => m.Name == targetMethod.Name);
```

Comment:
`m.Name == targetMethod.Name` is not sufficient. In case when there are two methods with the same name but different parameters, this will match the wrong method. For example:
```cs
public interface IFoo
{
    void Hello(string s);
    void Hello(int n);
}
```

Reply:
@cnshenj I agree. We must do better than this. Maybe it is time to drop netstandard1.1 support anyways. 😀

Reply:
Actually we have on `TypeInfo`. It just has another name. `GetRuntimeInterfaceMap` 😀

Reply:
Good point. I added tests that confirm `GetRuntimeInterfaceMap` does work. See PR #27.

Also, I'm wondering why you keep using GetTypeInfo() in places when `Type` is sufficient. For example, `proxyDefinition.TargetType.GetTypeInfo().IsInterface` can be simplified as `proxyDefinition.TargetType.IsInterface`. And `Type.IsInterface` probably has slightly better performance since it saves one method call.